frontend/src/components/App/Notifications/index.tsx

```diff
@@ -0,0 +1,226 @@
+import bellIcon from '@iconify/icons-mdi/bell';
+import { Icon } from '@iconify/react';
+import { Badge, Grid, IconButton, makeStyles, Theme, Tooltip, Typography } from '@material-ui/core';
+import { Menu, MenuItem } from '@material-ui/core';
+import { useEffect, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { useDispatch } from 'react-redux';
+import { useHistory } from 'react-router';
+import { FixedSizeList } from 'react-window';
+import Event, { KubeEvent } from '../../../lib/k8s/event';
+import { Notification } from '../../../lib/notification';
+import { createRouteURL } from '../../../lib/router';
+import { setUINotifications, updateUINotification } from '../../../redux/actions/actions';
+import { useTypedSelector } from '../../../redux/reducers/reducers';
+import { DateLabel } from '../../common';
+
+const useStyles = makeStyles((theme: Theme) => ({
+  notificationItem: {
+    '&.MuiMenuItem-root': {
+      borderBottom: `1px solid ${theme.palette.notificationBorderColor}`,
+      padding: '1rem',
+    },
+  },
+  notificationMessage: {
+    '&': {
+      wordBreak: 'break-word',
+      whiteSpace: 'normal',
+    },
+  },
+  root: {
+    '& .MuiPaper-root': {
+      width: '30vw',
+      minWidth: '300px',
+      maxHeight: '70vh',
+    },
+  },
+  errorItem: {
+    '&': {
+      color: theme.palette.error,
+    },
+  },
+}));
+
+function prepareNotificationsList(
+  notifications: Notification[],
+  clickEventHandler: (notification?: Notification, closeMenu?: boolean) => void
+) {
+  const { t } = useTranslation();
+  const classes = useStyles();
+  const history = useHistory();
+  if (!notifications || notifications.length === 0) {
+    return (
+      <MenuItem onClick={() => clickEventHandler()} className={classes.notificationItem} disabled>
+        {t(`frequent|You don't have any notifications right now`)}
+      </MenuItem>
+    );
+  }
+
+  function notificationSeenUnseenHandler(event: any, notification?: Notification) {
+    event.stopPropagation();
+    if (!notification) {
+      return;
+    }
+    notification.seen = true;
+    clickEventHandler(notification);
+  }
+
+  function notificationItemClickHandler(notification: Notification) {
+    notification.url && history.push(notification.url);
+    notification.seen = true;
+    clickEventHandler(notification, true);
+  }
+
+  function Row(props: { index: number; style: object }) {
```

Comment:
nit: I saw here the type for props: https://mui.com/material-ui/react-list/#virtualized-list

```ts
import { FixedSizeList, ListChildComponentProps } from 'react-window';
function renderRow(props: ListChildComponentProps) {
```

Reply:
thank you, fixed